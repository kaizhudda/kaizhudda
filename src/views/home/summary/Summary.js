import React, { useEffect } from 'react';
import Title from '../../../sharedComponents/title';
import Subtitle from '../../../sharedComponents/subtitle';

import './Summary.scss';

const Summary = ({ fetchSummary, summaryData }) => {
  useEffect(() => {
    fetchSummary();
  }, [fetchSummary])

  return (
    <div className="Summary row center-xs">
      <Title>Profile</Title>
      <Subtitle>
        {
          summaryData &&
          summaryData.fields.subtitle
        }
      </Subtitle>
      <div className="profile-content col-xl-5 col-lg-6 col-md-9 col-sm-11 col-xs-12">
        <div className="picture" />
        <p className="description">
          {
            summaryData &&
            summaryData.fields.description
          }
          <br /> <br />
          <i className="fa fa-map-marker-alt" /> <span className="location">
            {
              summaryData &&
              summaryData.fields.location
            }
          </span>
        </p>
      </div>
    </div>
  );
}

export default Summary;
