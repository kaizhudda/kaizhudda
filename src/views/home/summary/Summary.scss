@import '../../../styles/variables';
@import '../../../styles/media-queries';

.Summary {
  background: white;
  color: #727878;
  font-family: $font-body;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 16px 16px 48px 16px;

  .profile-content {
    display: flex;
    flex-direction: column;

    .description {
      line-height: 1.5;
      font-size: 1.125rem;

      @include phone-only {
        font-size: 1rem;
      }

      .location {
        color: #22a39f;
      }

      .fa {
        color: #22a39f;
      }
    }

  }


  .button {
    padding: 12px 28px;
    border: 1px solid transparent;
    background: #f56e4a;
    color: white;
    font-weight: 500;
    transition: all 0.5s cubic-bezier(.25,.8,.25,1);
    text-transform: capitalize;
    font-size: 14px;
    text-decoration: none;
    align-self: center;
    cursor: pointer;
    letter-spacing: 1px;
    text-transform: uppercase;
    border-radius: 20px;

    &:hover {
      background: #f45b32;
      color: white;
    }

    @include phone-only {
      padding: 9px 20px;
      font-size: 11px;
    }
  }
}