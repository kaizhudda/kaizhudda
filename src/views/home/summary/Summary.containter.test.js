import React from 'react';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import Summary from './index';

let wrapped;
const mockStore = configureMockStore([thunk]);
const intitalState = {
  summary: [{ fields: { subtitle: '', description: '', location: '' } }]
}
const props = {
  summaryData: { fields: { subtitle: 'sub', description: 'desc', location: 'local' } }
}

describe('Summary', () => {
  beforeEach(() => {
    const store = mockStore(intitalState);
    wrapped = mount(<Summary store={store} {...props} />);
  });

  it('should render the Summary Container Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should contain summaryData prop', () => {
    expect(wrapped.props().summaryData).toBe(props.summaryData);
  });
});
