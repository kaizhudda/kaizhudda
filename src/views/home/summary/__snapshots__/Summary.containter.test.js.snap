// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary should render the Summary Container Component correctly 1`] = `
<Connect(Summary)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  summaryData={
    Object {
      "fields": Object {
        "description": "desc",
        "location": "local",
        "subtitle": "sub",
      },
    }
  }
>
  <Summary
    fetchSummary={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    summaryData={
      Object {
        "fields": Object {
          "description": "",
          "location": "",
          "subtitle": "",
        },
      }
    }
  >
    <div
      className="Summary row center-xs"
      id="summary-section"
    >
      <Title>
        <h1
          className="Title "
        >
          Profile
        </h1>
      </Title>
      <Subtitle>
        <h3
          className="Subtitle"
        />
      </Subtitle>
      <div
        className="profile-content col-xl-5 col-lg-6 col-md-9 col-sm-11 col-xs-12"
      >
        <div
          className="picture"
        />
        <p
          className="description"
        >
          <br />
           
          <br />
          <i
            className="fa fa-map-marker-alt"
          />
           
          <span
            className="location"
          />
        </p>
      </div>
    </div>
  </Summary>
</Connect(Summary)>
`;
